test/libsolidity/util/BytesUtils.cpp

```diff
@@ -159,13 +158,8 @@ string BytesUtils::formatBoolean(bytes const& _bytes)
 
 string BytesUtils::formatHex(bytes const& _bytes)
 {
-	stringstream os;
-
-	string hex{toHex(_bytes, HexPrefix::Add)};
-	boost::algorithm::replace_all(hex, "00", "");
-	os << hex;
-
-	return os.str();
+	u256 value = fromBigEndian<u256>(_bytes);
```

Comment:
Why is `_bytes` being cropped to `u256.` Is it the case that input to `formatHex` is at most 32 bytes?

Reply:
Please at least add a check about the length of `_bytes`. What happens if it is shorter than 32 bytes and starts with zeros?

Reply:
@bshastry Yes, it's supposed to be called with at most 32 bytes. I added an assertion for that.
@chriseth It's being cropped: e.g. `bytes{0,0,0,3,255,0,2}` becomes `0x3FF0002`

Reply:
@erak if that's what we want then it's fine.

Reply:
I think so. But I'm also fine with asserting that this function is only called with an input of 32 bytes. Unpadded values should be handled as hex strings in any case.